l10n_it_dichiarazione_intento/tests/test_dichiarazione_intento.py

```diff
@@ -237,3 +237,12 @@ def test_refund_with_amount_bigger_than_residual(self):
     def test_fiscal_position_no_dichiarazione(self):
         self.invoice4._onchange_date_invoice()
         self.assertEqual(self.invoice4.fiscal_position_id.id, self.fiscal_position2.id)
+
+    def test_multiple_invoices(self):
+        """Check that validating multiple invoices
+        when there is no valid declaration raises an error."""
+        self.dichiarazione1.unlink()
+        invoices = self.invoice1 | self.invoice2
+        with self.assertRaises(UserError) as ue:
+            invoices.action_invoice_open()
+        self.assertIn('not found', ue.exception.name)
```

Comment:
potresti verificare piuttosto l'uguaglianza del messaggio dell'eccezione?

Reply:
Grazie della review!
Quindi nel test vorresti verificare che tutta la stringa
```
'Declaration of intent not found. '
'Add new declaration or '
'change fiscal position and verify applied tax'
```
è nell'eccezione sollevata?
Non mi pare una buona idea perché la stringa dell'eccezione potrebbe essere migliorata/modificata (ad esempio inserendo il numero della fattura) e non credo sia una cosa da tenere sotto controllo tramite test automatico.
Va bene se lascio così o secondo te è bloccante per il merge?

Ho aggiunto l'`assert` su `'not found'` perché il concetto di base dell'errore è che la dichiarazione non è stata trovata quindi mi pare ragionevole che nell'eccezione ci sia un `'not found'`, e poi per evitare di avere un falso positivo nel caso in cui venga sollevato un altro errore tipo `'Expected singleton'`.

Reply:
> Va bene se lascio così o secondo te è bloccante per il merge?

Non ho capito se secondo te è una modifica bloccante per il merge, potresti rispondere alla domanda qui sopra?
(viene da https://github.com/OCA/l10n-italy/pull/2548#discussion_r767550161)

> Puoi aggiungere almeno `Declaration of intent not found` nella stringa ricercata?

Questo mi sembra più ragionevole, fatto grazie 🙂